### ISSUE TITLE
Add documentation for APIML Spring Profiles

### ISSUE THREAD

**Issue by struga0258** [zowe/docs-site#1981]
The apiml squad hasn’t ever written docs about this explicitly. In the past we relied on creating new variables and putting them into instance.env then capturing those variable names in our start up scripts (e.g. -Dspring.profiles.active=${APIML_SPRING_PROFILES} so the user sets APIML_SPRING_PROFILES instead of SPRING_PROFILES_ACTIVE.
Recently we’ve started cutting out that middle-man variable, so in the above example we would tell users just to set SPRING_PROFILES_ACTIVE in instance.env , but we haven’t documented that this is possible for any variable, just documenting what the variable names are. It would make sense to only add some kind of documentations so that customers are aware of it and link to the spring docs on how spring works.

**Comment by anton-brezina** [comment]
Hi, @struga0258 , do you consider this relevant for v2 as the instance.env won't be available there?

**Comment by struga0258** [comment]
Yes, even though instance.env wont exist anymore zowe.yaml will. I have been told that this feature will still work under zowe.yaml and we need some sort of documentation, so customer are aware of this feature and know how to use it. We dont have to go into details about spring and all the spring functionality, we can just link spring docs but at least add some sort of documentation explaining the feature.

**Comment by struga0258** [comment]
This functionality was added because customers were having issues with security scans and needed to change their headers, it would make sense to allow them to be able to have this feature specially since it works in v1. Most important add some documentation just on how it works from the Zowe end.

**Comment by balhar-jakub** [comment]
I have been going through the usage of this parameter and there are following used profiles:
- attls
- debug
- diag
- dev

From these profiles
- attls is documented here: https://docs.zowe.org/stable/user-guide/api-mediation/api-mediation-internal-configuration/#at-tls
- debug is used when debug mode is enabled on Zowe level via LOG_LEVEL property
- diag and dev are internal development profiles that aren't documented intentionally. 

Based on the discussion around the addition of headers required by some security scans. This isn't happening via APIML_SPRING_PROFILES. There is a separate part in the documentation https://docs.zowe.org/stable/extend/extend-apiml/custom-metadata/

This is the excerpt from the documentation.

customMetadata.apiml.response.headers
(Optional) A service can specify headers that are added to the response by the Gateway. When this parameter is not set or is empty, no headers are added. Header names and header values are separated by :. Multiple headers can be added, delimited by ,. If a header with the same name already exists in the response, the Gateway overwrites the value of the header.
  Examples:
Strict-Transport-Security:max-age=1234; includeSubDomains
Sets a header with name Strict-Transport-Security and value max-age=1234; includeSubDomains.
Strict-Transport-Security:max-age=1234; includeSubDomains, X-Frame-Options:SAMEORIGIN
Sets two headers:
1) Header with name Strict-Transport-Security and value max-age=1234; includeSubDomains. 2) Header with name X-Frame-Options and value `SAMEORIGIN.

@CarsonCook Do you know about any other usage of spring profiles that I missed?

**Comment by CarsonCook** [comment]
@balhar-jakub no I am not aware of any other profiles
